src/engine/strat_engine/backstore/identify.rs

```diff
@@ -118,6 +188,55 @@ fn device_identifiers_wrapper(
         })
 }
 
+/// Process a device which udev information indicates is a LUKS device.
+fn process_luks_device(dev: &libudev::Device) -> Option<LuksInfo> {
+    match dev.devnode() {
+        Some(devnode) => match device_to_devno_wrapper(dev) {
+            Err(err) => {
+                warn!(
+                    "udev identified device {} as a Stratis device but {}, disregarding the device",
+                    devnode.display(),
+                    err
+                );
+                None
+            }
+            Ok(device_number) => match CryptHandle::setup(devnode) {
+                Ok(None) => None,
+                Err(err) => {
+                    warn!(
+                            "udev identified device {} as a LUKS device, but could not read LUKS header from the device, disregarding the device: {}",
+                            devnode.display(),
+                            err,
+                            );
+                    None
+                }
+                Ok(Some(handle)) => {
+                    match KeyDescription::from_system_key_desc(handle.key_description()) {
```

Comment:
I looked into it and when I reviewed your changes to my key management PR, I thought you had also changed `CryptHandle` to use `KeyDescription`. It seems that I missed that and I think it really should. Based on this, I think it would be good to change this to `KeyDescription` instead of a `String` so that either can be accessed.

Reply:
I changed the ```CryptInitializer``` methods to take a ```KeyDescription``` as a parameter. I was more hesitant with ```CryptHandle``` for what are probably trifling semantic reasons. I'll go ahead and see how that goes.